slackbot_using_azure_openai/service.bal

```diff
@@ -0,0 +1,88 @@
+import ballerina/crypto;
+import ballerina/http;
+import ballerinax/azure.openai.chat;
+
+configurable string openAIToken = ?;
+configurable string serviceUrl = ?;
+configurable string deploymentId = ?;
+configurable string slackSigningSecret = ?;
+
+const API_VERSION = "2023-03-15-preview";
+const MAX_MESSAGES = 25;
+
+final chat:Client azureOpenAI = check new (
+    config = {auth: {apiKey: openAIToken}},
+    serviceUrl = serviceUrl
+);
+
+type ChatMessage record {|
+    string role;
+    string content;
+|};
+
+type Response record {|
+    string response_type;
+    string text;
+|};
+
+enum ROLE {
+    SYSTEM = "system",
+    USER = "user",
+    ASSISTANT = "assistant"
+}
+
+service /slack on new http:Listener(8080) {
+    map<ChatMessage[]> chatHistory = {};
+
+    resource function post events(http:Request request) returns Response|error {
+        if !check verifyRequest(request) {
+            return error("Request verification failed");
+        }
+
+        map<string> requestPayload = check request.getFormParams();
+
+        string? channelName = requestPayload["channel_name"];
+        string? requestText = requestPayload["text"];
+        if channelName is () || requestText is () {
+            return error("Invalid values in the request payload for channel_name or text");
```

Comment:
```suggestion
            return error("Invalid values in the request parameters for channel_name or text");
```

Reply:
addressed in [e4860cc](https://github.com/ballerina-guides/ai-samples/pull/64/commits/e4860cc5470355fbe328b8c4f811359070e2b88e)